onecodex/viz/_metadata.py

```diff
@@ -151,6 +158,21 @@ def plot_metadata(
             df = pd.concat([df, vert_df], axis=1).dropna(subset=[vert_magic_fields[vaxis]])
             magic_fields.update(vert_magic_fields)
 
+        if filter_samples_missing_abundances and self._all_nan_classification_ids:
```

Comment:
```suggestion
        if filter_samples_missing_abundances and self._all_nan_classification_ids:
```

I think `_all_nan_classification_ids` will always be empty if you use a metric other than abundance but maybe it's clearer to explicitly also check that the metric is `Abundance`.

It might also be clearer to rename `_all_nan_classification_ids` to `_classification_ids_without_abundances`

This is mostly just me pontificating and not a big concern